src/AspNetCoreModuleV2/CommonLib/debugutil.cpp

```diff
@@ -44,21 +46,54 @@ DebugInitialize()
     const size_t environmentVariableValueSize = 2;
     std::wstring environmentVariableValue(environmentVariableValueSize, '\0');
 
-    if (GetEnvironmentVariable(L"ASPNETCORE_MODULE_DEBUG", environmentVariableValue.data(), environmentVariableValueSize) == environmentVariableValueSize - 1)
+    try
     {
-        try
-        {
-            const auto value = std::stoi(environmentVariableValue);
+        const auto value = std::stoi(Environment::ExpandEnvironmentVariables(L"%ASPNETCORE_MODULE_DEBUG%"));
 
-            if (value >= 1) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_ERROR;
-            if (value >= 2) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_WARNING;
-            if (value >= 3) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_INFO;
-            if (value >= 4) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_CONSOLE;
-        }
-        catch (...)
+        if (value >= 1) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_ERROR;
+        if (value >= 2) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_WARNING;
+        if (value >= 3) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_INFO;
+        if (value >= 4) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_CONSOLE;
+    }
+    catch (...)
+    {
+        // ignore
+    }
+
+    try
+    {
+        const auto debugOutputFile = Environment::ExpandEnvironmentVariables(L"%ASPNETCORE_MODULE_DEBUG_FILE%");
+
+        if (!debugOutputFile.empty())
         {
-            // ignore
+            g_logFile = CreateFileW(debugOutputFile.c_str(),
+                FILE_GENERIC_WRITE,
+                FILE_SHARE_READ | FILE_SHARE_WRITE,
+                nullptr,
+                OPEN_ALWAYS,
+                FILE_ATTRIBUTE_NORMAL,
```

Comment:
FILE_APPEND_DATA so you don't need to always call SetFilePointer?

Reply:
FILE_APPEND_DATA  is part of FILE_GENERIC_WRITE you still need to set it to append to existing file.

Reply:
Then can you remove the SetFilePointer?

Reply:
It's required to append to an existing file, see https://msdn.microsoft.com/en-us/library/windows/desktop/aa363778(v=vs.85).aspx

Reply:
I seeeeeeeeeeeeeeeeeeeeeeeeeeeeee.

Reply:
you allow shared write which could cause file corrupt. should we just use env  variable for debug file folder and then create a temp file for each process instead of mixing the log from different processes. This will also help user to easily parse the log

Reply:
We would have to create multiple files per process then, because inproc and shim initialize their own logging, allowing shared writes allows to reuse the same file.

Reply:
you also need to make sure DebugInitialize will only initialize once. Otherwise, g_logFile will be reassigned and lead to handle leak.

Reply:
I'll move it into DLL_ATTACH